app/dashboard/src/hooks/backendHooks.ts

```diff
@@ -561,3 +617,95 @@ export function useBackendGetOrganization(backend: Backend | null) {
     uploadOrganizationPictureVariables,
   ])
 }
+
+// ==================================
+// === useBackendMutationListener ===
+// ==================================
+
+/** Listen to, and process, all mutation events. */
+export function useBackendMutationListener(backendType: backendModule.BackendType) {
```

Comment:
Well, it's not worth it. Optimistic updates should never be the solution, especially when we mutate on `success`. Let's simply do the invalidations and don't hide potential invalidation issues behind the optimistic updates.

If we want to add optimistic updates to the mutations, we can simply pass client as a prop to the mutation options builders.